src/mscorlib/src/System/String.cs

```diff
@@ -411,7 +415,61 @@ private unsafe static bool EqualsHelper(String strA, String strB)
                 return (length <= 0);
             }
         }
-        
+
+        [System.Security.SecuritySafeCritical]  // auto-generated
+        [ReliabilityContract(Consistency.WillNotCorruptState, Cer.MayFail)]
+        private unsafe static bool OrdinalCompareSubstring(String strA, String strB, int length)
+        {
+            Contract.Requires(strA != null);
```

Comment:
Does it really help anything on x64 to pass in the `length` as argument? It will likely hurt perf on x86. I think it would be better if this method just takes two strings.